eclipse/src/points-snap.ts

```diff
@@ -0,0 +1,182 @@
+import { AnchorProvider } from "@coral-xyz/anchor";
+import {
+  CreatePositionEvent,
+  getMarketAddress,
+  InvariantEventNames,
+  IWallet,
+  Market,
+  Network,
+  parseEvent,
+  RemovePositionEvent,
+} from "@invariant-labs/sdk-eclipse";
+import {
+  ConfirmedSignatureInfo,
+  Connection,
+  ParsedTransactionWithMeta,
+  PublicKey,
+} from "@solana/web3.js";
+import * as fs from "fs";
+
+enum EventTypes {
+  OpenPosition,
+  RemovePosition,
+}
+
+const fetchAllSignatures = async (
+  connection: Connection,
+  programId: PublicKey,
+  lastTxHash: string | undefined
+) => {
+  const allSignatures: ConfirmedSignatureInfo[] = [];
+  let beforeTxHash: string | undefined = undefined;
+  let done: boolean = false;
+
+  while (!done) {
+    const signatures = await connection.getSignaturesForAddress(
+      programId,
+      { before: beforeTxHash, until: lastTxHash },
+      "confirmed"
+    );
+
+    if (signatures.length === 0) {
+      done = true;
+      break;
+    }
+
+    allSignatures.push(...signatures);
+    if (lastTxHash === undefined) {
+      done = true;
+      break;
+    }
+    if (signatures[signatures.length - 1].signature === lastTxHash) {
+      done = true;
+    } else {
+      beforeTxHash = signatures[signatures.length - 1].signature;
+    }
+  }
+
+  return allSignatures.map((signatureInfo) => signatureInfo.signature);
+};
+
+const processParsedTransactions = (
+  parsedTransactions: (ParsedTransactionWithMeta | null)[]
+) => {
+  return parsedTransactions
+    .filter((tx) => tx?.meta?.logMessages && tx.transaction.signatures[0])
+    .map((tx) => {
+      return tx!.meta!.logMessages!;
+    });
+};
+
+const fetchTransactionLogs = async (
+  connection: Connection,
+  signatures: string[],
+  batchSize: number
+) => {
+  const batchCount = Math.ceil(signatures.length / batchSize);
+  const batchedSignatures = new Array(batchCount).fill(0);
+
+  return (
+    await Promise.all(
+      batchedSignatures.map(async (_, idx) => {
+        const batchSignatures = signatures.slice(
+          idx * batchSize,
+          (idx + 1) * batchSize
+        );
+        return processParsedTransactions(
+          await connection.getParsedTransactions(batchSignatures, "confirmed")
+        );
+      })
+    )
+  ).flat();
+};
+
+const extractEvents = (
```

Comment:
After parsing an event you should check program id that emitted event

Reply:
we gather tx hashes from program id, and then analyze program data, so it is event needed?